### WEB Scraping 

- sometimes we don't have an API (some websites don't provide API ) but we still need the data from that website then we use WEB SCRAPING 

- we will be feetching data from " Ambitionbox website " a list of company datas
- https://www.ambitionbox.com/list-of-companies?campaign=homepage_companies_widget&page=1
- the page= denotes the page nuber there are a total of 333 pages in the website in the list of companies

did  " pip list "  Beautifoul soup exists already

We will extract the following Details of the companies from each page
- Name, Rating, Public, HQ location, How Old company, No of Employees, Tags reltd to company, About(the company)
- so 9 columns , no of rows - as many companies

In [1]:
import pandas as pd
import requests
from bs4 import BeautifulSoup

In [2]:
requests.get('https://www.ambitionbox.com/list-of-companies?campaign=homepage_companies_widget&page=1')

<Response [403]>

- 403 Forbidden - The request was a legal request, but the server is refusing to respond to it

In [3]:
requests.get('https://www.ambitionbox.com/list-of-companies?campaign=homepage_companies_widget&page=1').text

'<HTML><HEAD>\n<TITLE>Access Denied</TITLE>\n</HEAD><BODY>\n<H1>Access Denied</H1>\n \nYou don\'t have permission to access "http&#58;&#47;&#47;www&#46;ambitionbox&#46;com&#47;list&#45;of&#45;companies&#63;" on this server.<P>\nReference&#32;&#35;18&#46;36f856b8&#46;1678951902&#46;3f053d5\n</BODY>\n</HTML>\n'